pkg/workers/send_work.go

```diff
@@ -0,0 +1,263 @@
+package workers
+
+import (
+	"fmt"
+	"time"
+
+	masa "github.com/masa-finance/masa-oracle/pkg"
+	"github.com/masa-finance/masa-oracle/pkg/pubsub"
+	"github.com/masa-finance/masa-oracle/pkg/workers/messages"
+
+	"github.com/asynkron/protoactor-go/actor"
+
+	pubsub2 "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/sirupsen/logrus"
+)
+
+var workerConfig *WorkerConfig
+
+func init() {
+	var err error
+	workerConfig, err = LoadConfig()
+	if err != nil {
+		logrus.Fatalf("Failed to load worker config: %v", err)
+	}
+	workerDoneCh = make(chan *pubsub2.Message, workerConfig.WorkerBufferSize)
+}
+
+type Worker struct {
+	IsLocal  bool
+	NodeData pubsub.NodeData
+	IPAddr   string
+	Node     *masa.OracleNode
+}
+
+func SendWork(node *masa.OracleNode, m *pubsub2.Message) {
+	logrus.Infof("Sending work to node %s", node.Host.ID())
+	props := actor.PropsFromProducer(NewWorker(node))
+	pid := node.ActorEngine.Spawn(props)
+	message := createWorkMessage(m, pid)
+
+	responseCollector := make(chan *pubsub2.Message, 1)
+
+	eligibleWorkers := GetEligibleWorkers(node, message)
+
+	success := tryWorkersRoundRobin(node, eligibleWorkers, message, responseCollector)
+	if !success {
+		logrus.Error("Failed to process the work")
+	}
+}
+
+func tryWorkersRoundRobin(node *masa.OracleNode, workers []Worker, message *messages.Work, responseCollector chan *pubsub2.Message) bool {
+	var localWorker *Worker
+	remoteWorkersAttempted := 0
+
+	logrus.Info("Starting round-robin worker selection")
+
+	// Try remote workers first, up to MaxRemoteWorkers
+	for _, worker := range workers {
```

Comment:
I thought we had some sort of LRU cache around here when I was thinking at this from the conversations - wouldn't make things easier to track? for instance, the node could track to which worker it sent already requests too, and try to schedule work to other workers first, and if it fails trying to piggy back at the last worker that completed successfully the job already

Reply:
yeah I think we now moved in this direction will request another review tomorrow standby

Reply:
For this release we are randomly selecting a remote worker node that we can connect to. Next iteration, I agree that adding a more comprehensive approach such as LRU will be added.